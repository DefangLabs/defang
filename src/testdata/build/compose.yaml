services:
  build1:
    build:
      context: .
      dockerfile: Dockerfile
      shm_size: 2147483648
      target: test
      args:
        - BASE_IMAGE=nginx
  build2:
    build:
      context: .
      dockerfile: Dockerfile
      shm_size: 2gb
      target: test
      args:
        BASE_IMAGE: alpine
