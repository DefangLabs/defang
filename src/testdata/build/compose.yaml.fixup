{
  "build1": {
    "build": {
      "context": ".",
      "dockerfile": "Dockerfile",
      "args": {
        "BASE_IMAGE": "nginx"
      },
      "target": "test",
      "shm_size": "2147483648"
    },
    "command": null,
    "entrypoint": null,
    "networks": {
      "default": null
    }
  },
  "build2": {
    "build": {
      "context": ".",
      "dockerfile": "Dockerfile",
      "args": {
        "BASE_IMAGE": "alpine"
      },
      "target": "test",
      "shm_size": "2147483648"
    },
    "command": null,
    "entrypoint": null,
    "networks": {
      "default": null
    }
  },
  "build3": {
    "build": {
      "context": ".",
      "dockerfile": "Dockerfile"
    },
    "command": null,
    "entrypoint": null,
    "networks": {
      "default": null
    }
  }
}