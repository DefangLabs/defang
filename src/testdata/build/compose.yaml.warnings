 ! service "build1": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "build2": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 * Packaging the project files for build1 at .
 * Packaging the project files for build2 at .