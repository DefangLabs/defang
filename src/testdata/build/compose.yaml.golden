name: build
services:
  build1:
    build:
      context: .
      dockerfile: Dockerfile
      args:
        BASE_IMAGE: nginx
      target: test
      shm_size: "2147483648"
    networks:
      default: null
  build2:
    build:
      context: .
      dockerfile: Dockerfile
      args:
        BASE_IMAGE: alpine
      target: test
      shm_size: "2147483648"
    networks:
      default: null
networks:
  default:
    name: build_default
